canvas_sdk/v1/data/base.py

```diff
@@ -35,13 +36,51 @@ def find(self, value_set: Type["ValueSet"]) -> models.QuerySet:
 
         Condition.objects.find(MorbidObesity).find(AnaphylacticReactionToCommonBakersYeast)
         """
-        values_dict = value_set.values
-        uri_codes = [
-            (i[1], values_dict[i[0]])
-            for i in value_set.CODE_SYSTEM_MAPPING.items()
-            if i[0] in values_dict
-        ]
         q_filter = Q()
-        for system, codes in uri_codes:
-            q_filter |= Q(codings__system=system, codings__code__in=codes)
+        for system, codes in self.codings(value_set):
+            q_filter |= Q(**self.q_kwargs(system, codes))
         return self.filter(q_filter).distinct()
+
+    @staticmethod
+    def codings(value_set: Type["ValueSet"]) -> tuple[tuple[str, set[str]]]:
+        """Provide a sequence of tuples where each tuple is a code system URL and a set of codes."""
+        values_dict = value_set.values
+        return cast(
+            tuple[tuple[str, set[str]]],
+            tuple(
+                (i[1], values_dict[i[0]])
+                for i in value_set.CODE_SYSTEM_MAPPING.items()
+                if i[0] in values_dict
+            ),
+        )
+
+    @staticmethod
+    def q_kwargs(system: str, codes: Container[str]) -> dict[str, str | Container[str]]:
+        """
+        This method can be overridden if a Q object with different filtering options is needed.
+        """
+        return {"codings__system": system, "codings__code__in": codes}
+
+
+class ValueSetLookupByNameQuerySet(ValueSetLookupQuerySet):
+    """
+    QuerySet for ValueSet lookups using code system name rather than URL.
+
+    Some models, like Questionnaire, store the code system by name (e.g. "LOINC") rather than by the
+    url (e.g. "http://loinc.org"). This subclass accommodates these models.
+    """
+
+    @staticmethod
+    def codings(value_set: Type["ValueSet"]) -> tuple[tuple[str, set[str]]]:
+        """
+        Provide a sequence of tuples where each tuple is a code system name and a set of codes.
+        """
+        values_dict = value_set.values
+        return cast(
+            tuple[tuple[str, set[str]]],
+            tuple(
+                (i[0], values_dict[i[0]])
```

Comment:
The only difference between this and the parent class is that we're using `i[0]` (the code system name) instead of `i[1]` (the code system URL). Questionnaire objects store the code system name rather than the URL.